src/ring_theory/unique_factorization_domain.lean

```diff
@@ -1136,6 +1173,18 @@ begin
   simpa only
 end
 
+theorem count_ne_zero_iff_dvd {a p : α} (ha0 : a ≠ 0) (hp : irreducible p) :
+  (associates.mk p).count (associates.mk a).factors ≠ 0 ↔ p ∣ a :=
```

Comment:
I have a vague idea it might be nicer to avoid using `associates.mk`, since then you wouldn't have to do a case split if you already have `a p : associates α`. On the other hand, with the change you'll have to rewrite with `associates.mk_ne_zero`/`associates.irreducible_mk` if you want to apply this for `a p : α`. Do you have a use in mind where `associates.mk` is more convenient?

In the end I'll let you choose which form is better.
```suggestion
theorem count_ne_zero_iff_dvd {a p : associates α} (ha0 : a ≠ 0) (hp : irreducible p) :
  p.count a.factors ≠ 0 ↔ p ∣ a :=
```